### PR TITLE
Slightly improve usability issues with print

### PR DESCRIPTION
**Pull request by tsauerwein**
- Shows a loading spinner during the print process.
 - Directly starts the download once the job is finished.
 -  ~~Fixes visual glitches (The panel showing the progress is cleared when a new job is started.).~~ (Apparently the "visual glitches" are the expected behavior.)

### CODE REVIEW

File: geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js [hunk 1]
```diff
@@ -429,10 +430,12 @@ GeoExt.ux.SimplePrint = Ext.extend(Ext.form.FormPanel, {
      *  Do the print.
      */
     print: function() {
+        // TODO cancel ongoing print job if any
```
Comment (sbrunner) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: Why cancel previous job ?

File: geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js [hunk 2]
```diff
@@ -429,10 +430,12 @@ GeoExt.ux.SimplePrint = Ext.extend(Ext.form.FormPanel, {
...
         if (this.printProvider.supportProgress()) {
             var self = this;
             var interval = null;
             var updateStatus;
+            this.progressPanel.removeAll();
```
Comment (sbrunner) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: Why remove all ? what for the previous job ?

File: geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js [hunk 3]
```diff
@@ -472,16 +477,12 @@ GeoExt.ux.SimplePrint = Ext.extend(Ext.form.FormPanel, {
...
-                        self.progressPanel.doLayout();
-                        self.progressPanel.setHeight(35 * self.progressPanel.items.length);
-                        self.doLayout();
                         return false;
                     };
-                }
-                else {
+                    self.printProvider.download(
+                        self.printProvider.getDownloadURL(job)
+                    );
```
Comment (sbrunner) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: For me it's not coherent, you directly do the download and add an URL ?
If we do a directly download we also should remove the panel from the queue...

File: geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js (hunk 1 above)
Comment (tsauerwein) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: If the user hits "Print" again, before the previous job has finished, you want to cancel the previous job.

Comment (sbrunner) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: No you want the booth prints...

File: geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js (hunk 2 above)
Comment (tsauerwein) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: I think I now understand what you originally wanted to do. You want that the user can submit multiple print requests and that there is a list with all the ongoing and finished requests. This wasn't communicated very well in the UI. For me the UI simply looked broken.

Is this required by the spec that the user can submit multiple print requests at the same time, or should a previous print request be cancelled if the user hits "Print" again? @ybolognini ?

File: geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js (hunk 3 above)
Comment (tsauerwein) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: I am totally ok with removing the download link if the download is started automatically. I just thought that you had a good reason to show the download link in the first place.

Comment (ybolognini) [inline geoext.ux/ux/SimplePrint/lib/GeoExt.ux/SimplePrint.js]: Yes, removing the Download link is what we want.
As for the fact to cancel the previous job when clicked a second time on Print, I would prefer to launch two jobs. @sbrunner told me it wasn't hard technically, so I think it's the best: direct download AND possible to launch more than one prints/downloads.

**Comment by tsauerwein** [comment]
Updated PR

**Comment by sbrunner** [comment]
Looks good for me, you can merge when Travis tests pass :-)

**Pull request merged**